### PR TITLE
Add support for ledger and trezor with bitcoin-core/HWI

### LINKED ISSUE #36: Bitcoin Hardware Wallet Integration
**Time Horizon:** Medium term

**Description:** Integrate hardware wallets with PSBT in Bitcoin Core 0.18.0

**Motivation:** Users may wish to use a hardware wallet with the hardware wallet functionality coming in Bitcoin Core 0.18.0 on top of the full-node that comes with nix-bitcoin.

**Possible Implementation:** Separate bitcoin service with wallet functionality.

### PR DESCRIPTION
**Pull request by jonasnick**
fixes #36

### CODE REVIEW

**Comment by nixbitcoin** [comment]
`hwi enumerate` works well with Trezor v1. Sadly, if the user has electrs enabled this makes `disablewallet=1` in `/var/lib/bitcoind/bitcoin.conf` and prevents using HWI with Bitcoin Core on the nix-bitcoin node. How about creating a custom, restricted bitcoin-cli for use with HWI?

Use of the electrum server and Bitcoin HWI is somewhat contradictory so it might be an option to just make this an either or.

File: configuration.nix [hunk 1]
```diff
@@ -78,6 +78,15 @@
   #   "djbooth007" = 1000;
   # };
 
+  ### Hardware wallets
+  # Enable this module to allow using hardware wallets. See https://github.com/bitcoin-core/HWI
+  # for more information.
+  # Ledger must be initialized through the official ledger live app and the Bitcoin app must
+  # be installed and running on the device.
+  # services.hardware-wallets.ledger = true;
+  # Trezor can be initialized with the trezorctl command in nix-bitcoin.
```
Comment (nixbitcoin) [inline configuration.nix]: This line and line 84-85 could be replaced by appropriate entries in the usage.md like #63

**Comment by jonasnick** [comment]
> Use of the electrum server and Bitcoin HWI is somewhat contradictory so it might be an option to just make this an either or.

For now making this an either or would be fine for me.

> How about creating a custom, restricted bitcoin-cli for use with HWI?

You mean bitcoind, not bitcoin-cli right? Yes, that's an option, but this has considerable overhead - especially if you don't use electrs.

File: configuration.nix (hunk 1 above)
Comment (jonasnick) [reply configuration.nix]: Imo for this PR it's fine. You can remove a part of that in #63 if you want.I'd like concise comments in configuration.nix - especially if it helps us - but in this case seems fine to move it to usage.md. Maybe in #63 you can remove these lines and append a line like: "More detailed information can be found in docs/usage.md"?

**Comment by jonasnick** [comment]
rebased

**Pull request merged**